### PR TITLE
openstackclient: add module

### PR DESCRIPTION
**Pull request by tensor5**
### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

File: modules/programs/openstackclient.nix [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  options.programs.openstackclient = {
+    enable = lib.mkEnableOption "OpenStack command-line client";
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.openstackclient;
+      defaultText = lib.literalExpression "pkgs.openstackclient";
+      description = "Package providing {command}`openstack`.";
+    };
```
Comment (sumnerevans) [inline modules/programs/openstackclient.nix:17]:
```suggestion
    package = lib.mkPackageOption pkgs "openstackclient";
```

File: modules/programs/openstackclient.nix [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        Configuration written to {file}`~/.config/openstack/clouds-public.yaml`.
+        See <https://docs.openstack.org/python-openstackclient/latest/configuration/index.html#clouds-public-yaml>.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+
+    home.file."${config.home.homeDirectory}/.config/openstack/clouds.yaml".source =
```
Comment (sumnerevans) [inline modules/programs/openstackclient.nix:68]:
```suggestion
    xdg.configFile."openstack/clouds.yaml".source =
```

File: modules/programs/openstackclient.nix [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      yamlFormat.generate
+      "openstackclient-clouds-yaml-${config.home.username}" {
+        clouds = cfg.clouds;
+      };
+
+    home.file."${config.home.homeDirectory}/.config/openstack/clouds-public.yaml".source =
```
Comment (sumnerevans) [inline modules/programs/openstackclient.nix:74]:
```suggestion
    xdg.configFile."openstack/clouds-public.yaml".source =
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]

File: modules/programs/openstackclient.nix [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            region_name = "XXX";
+            interface = "internal";
+          };
+        };
+      '';
+      description = ''
+        Configuration needed to connect to one or more clouds.
+
+        Do not include passwords here as they will be publicly readable in the Nix store.
+        Configuration written to {file}`~/.config/openstack/clouds.yaml`.
```
Comment (sumnerevans) [inline modules/programs/openstackclient.nix:34]:
```suggestion
        Configuration written to {file}`$XDG_CONFIG_HOME/openstack/clouds.yaml`.
```

File: modules/programs/openstackclient.nix [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+          my-infra = {
+            cloud = "example-cloud";
+            auth = {
+              project_id = "0123456789abcdef0123456789abcdef";
+              username = "openstack";
+            };
+            region_name = "XXX";
+            interface = "internal";
+          };
+        };
```
Comment (sumnerevans) [inline modules/programs/openstackclient.nix:28]:
```suggestion
        }
```

**Comment by sumnerevans** [comment]
once you've made all of your changes, please also squash into a single commit

**Review by sumnerevans — APPROVED** [review]
lgtm

**Comment by tensor5** [comment]
Can someone approve the workflow, please?

**Pull request merged**